src/rdbuf.h

```diff
@@ -147,7 +147,11 @@ static RD_INLINE RD_UNUSED size_t rd_buf_write_pos(const rd_buf_t *rbuf) {
  * @returns the number of bytes available for writing (before growing).
  */
 static RD_INLINE RD_UNUSED size_t rd_buf_write_remains(const rd_buf_t *rbuf) {
-        return rbuf->rbuf_size - (rbuf->rbuf_len + rbuf->rbuf_erased);
+        ssize_t remaining =
```

Comment:
This change can be reverted as this condition must never happen, the problem happens because a segment is removed without decreasing the corresponding part in `rbuf->rbuf_erased` , in `rd_buf_erase` it should be:
```c
                /* If segment is now empty, remove it */
                if (seg->seg_of == 0) {
                        rd_buf_destroy_segment(rbuf, seg);
                        rbuf->rbuf_erased -= toerase;
                }
``` 
The issue can happen with RPCs using flexver as happens with MetadataRequest since version 9.

Reply:
Updated